src/hdf5_dl.c

```diff
@@ -97,44 +105,209 @@ hid_t H5E_ERR_CLS_g = -1;
 
 static bool is_init = false;
 
+/*
+ * Try to find a symbol within a library
+ * 
+ * handle: Handle to the library
+ * symbol: Symbol to look for
+ * Returns: a pointer to the symbol or NULL 
+ * if the symbol can't be found 
+ */
+void *find_sym(void *handle, const char *symbol) {
+
+  void *ret = NULL, *err = NULL;
+  dlerror(); /* clear error code */
+  ret = dlsym(handle, symbol);
+
+  if(ret != NULL && (err = dlerror()) == NULL)
+    return ret;
+  else
+    return NULL;
+}
+
+/*
+ * Check that all symbols have been loaded
+ * 
+ * Returns: 0 if an error occured, -1 for success
+ */
+int check_symbols() {
+
+  int retval = -1;
```

Comment:
You don't need `retval`.  Short circuit the tests: use `return 0` instead `retval = 0` and then a `return -1` at the end to cover the successful case.